main.tf

```diff
@@ -208,7 +208,7 @@ locals {
 resource "aws_eks_access_entry" "this" {
   for_each = { for k, v in local.merged_access_entries : k => v if local.create }
 
-  cluster_name      = aws_eks_cluster.this[0].name
+  cluster_name      = aws_eks_cluster.this[0].id
```

Comment:
I suspect that on some resources, `id` != `name` logically, but I don't think we ever hit a case like this where one must be used over another. Good catch!

PS: Check your email, please :)